src/less/pill/ds4/pill.less

```diff
@@ -0,0 +1,194 @@
+@import "../../less/ds4/variables.less";
+@import "../../less/ds4/mixins.less";
+@import "../../mixins/button/ds4/button-mixins.less";
+@import "../base/pill.less";
+
+.pill-size-rules(@fontSize, @padding) {
+    font-size: @fontSize;
+    padding: @padding;
+}
+
+button.pill,
+a.fake-pill {
+    line-height: 1.3em;
+
+    &__cell--fixed-height &__icon {
+        align-self: center;
+        overflow: visible;
+    }
+
+    &--small {
+        .pill-size-rules(@ds4-font-size-12, 8px 32px);
+
+        &-fixed-height {
+            height: 32px;
+        }
+
+        &-truncated {
+            .pill-size-rules(@ds4-font-size-12, 8px 32px);
+            .btn-truncate;
+
+            height: 32px;
+        }
+    }
+
+    &--medium {
+        .pill-size-rules(@ds4-font-size-14, 11px 40px);
+
+        &-fixed-height {
+            height: 40px;
+        }
+
+        &-truncated {
+            .pill-size-rules(@ds4-font-size-14, 11px 40px);
+            .btn-truncate;
+
+            height: 40px;
+        }
+    }
+
+    &--large {
+        .pill-size-rules(@ds4-font-size-16, 14px 48px);
+
+        &-fixed-height {
+            height: 48px;
+        }
+
+        &-truncated {
+            .pill-size-rules(@ds4-font-size-16, 14px 48px);
+            .btn-truncate;
+
+            height: 48px;
+        }
+    }
+
+    &--extra-large {
+        // chrome and firefox disagree on calculations, so 15.875px is a compromise
+        .pill-size-rules(@ds4-font-size-18, 15.875px 56px);
+
+        &-fixed-height {
+            height: 56px;
+        }
+
+        &-truncated {
+            .pill-size-rules(@ds4-font-size-18, 15.875px 56px);
+            .btn-truncate;
+
+            height: 56px;
+        }
+    }
+}
+
+button.pill--regular,
+a.fake-pill--regular {
+    &:hover,
+    &:focus {
+        border-color: @ds4-color-core-gray-spanish;
+    }
+}
+
+button.pill--primary,
```

Comment:
Should we nest the properties here? I see in some places you did and some places didnt. Or is it better to be verbose?
Like: 
button.pill,
a.fake-pill {
   &--primary,
   &--secondary {
   }
}